src/query/service/src/pipelines/processors/transforms/hash_join/hash_join_build_state.rs

```diff
@@ -220,6 +220,15 @@ impl HashJoinBuildState {
                 }
             }
 
+            let build_chunks =
```

Comment:
We can first check if `build_num_rows` is less than 10k and then decide whether to store datablocks into `build_chunks`,

Reply:
For `build_num_rows` is more than 10k, we also want to generate filter, such as bloom filter

Reply:
We can add a check if join is broadcast join, usually, if broadcast join, build side is small

Reply:
> For `build_num_rows` is more than 10k, we also want to generate filter, such as bloom filter

But now we do not collect information during finalize, and we only support inlist filter now.

Reply:
> For `build_num_rows` is more than 10k, we also want to generate filter, such as bloom filter

If we collect information during finalize, we can later remove these checks.